### PR TITLE
Enable quick aot entry when GC is enabled

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

File: .github/workflows/spec_test_on_nuttx.yml [hunk 1]
```diff
@@ -34,13 +34,13 @@ jobs:
     with:
       os: "ubuntu-22.04"
       arch: "ARM RISCV AArch64"
-      container_image: ghcr.io/no1wudi/nuttx/apache-nuttx-ci-linux:latest
+      container_image: ghcr.io/no1wudi/nuttx/apache-nuttx-ci-linux:8c4e00b607d4d6d66ba8f51c4544819a616eac69d3a2ac669e2af2150e2eb0f9
```
Comment (no1wudi) [inline .github/workflows/spec_test_on_nuttx.yml:37]:
```suggestion
      container_image: ghcr.io/no1wudi/nuttx/apache-nuttx-ci-linux:@sha245:8c4e00b607d4d6d66ba8f51c4544819a616eac69d3a2ac669e2af2150e2eb0f9
```

File: .github/workflows/spec_test_on_nuttx.yml [hunk 2]
```diff
@@ -34,13 +34,13 @@ jobs:
...
 
   spec_test_on_qemu:
     runs-on: ubuntu-latest
     needs: [build_llvm_libraries]
     container:
-      image: ghcr.io/no1wudi/nuttx/apache-nuttx-ci-linux:latest
+      image: ghcr.io/no1wudi/nuttx/apache-nuttx-ci-linux:8c4e00b607d4d6d66ba8f51c4544819a616eac69d3a2ac669e2af2150e2eb0f9
```
Comment (no1wudi) [inline .github/workflows/spec_test_on_nuttx.yml:43]:
```suggestion
      image: ghcr.io/no1wudi/nuttx/apache-nuttx-ci-linux:@sha256:8c4e00b607d4d6d66ba8f51c4544819a616eac69d3a2ac669e2af2150e2eb0f9
```

**Comment by no1wudi** [comment]
Now it seems that this is an occasional error caused by the test script, try to retrigger the workflow again?

**Comment by wenyongh** [comment]
> Now it seems that this is an occasional error caused by the test script, try to retrigger the workflow again?

Failed to run test-guard-page-skip, not sure why:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/7553863769/job/20565649590
```bash
============> run skip-stack-guard-page failed with a non-zero return code 101

==================== LAST LOG of skip-stack-guard-page ====================


Testing(exhaustion) test-guard-page-skip 300 = Exception: call stack exhausted

THE FINAL EXCEPTION IS argument of type 'NoneType' is not iterable
Compiling '/tmp/tmpyh1l4lk5.wasm' to '/tmp/tmp7gmej11x.aot'```

**Comment by wenyongh** [comment]
> > Now it seems that this is an occasional error caused by the test script, try to retrigger the workflow again?
> 
> Failed to run test-guard-page-skip, not sure why: https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/7553863769/job/20565649590
> 
> ```shell
> ============> run skip-stack-guard-page failed with a non-zero return code 101
> 
> ==================== LAST LOG of skip-stack-guard-page ====================
> 
> 
> Testing(exhaustion) test-guard-page-skip 300 = Exception: call stack exhausted
> 
> THE FINAL EXCEPTION IS argument of type 'NoneType' is not iterable
> Compiling '/tmp/tmpyh1l4lk5.wasm' to '/tmp/tmp7gmej11x.aot'```
> ```

Run again, this CI runs successfully, but riscv32 CI failed:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/7553863769/job/20568614964
```
============> run unreachable failed with a non-zero return code 101

==================== LAST LOG of unreachable ====================

THE FINAL EXCEPTION IS Failed:
 Result 0 incorrect:
 expected: 'Exception: unreachable'
  got: 'Exception: native stack overflow'
```

**Comment by wenyongh** [comment]
@no1wudi I found the quick-aot-entry was not applied for aot when hw bound check isn't enabled, and the CIs run success when I fixed that with https://github.com/bytecodealliance/wasm-micro-runtime/pull/3015/commits/446f4dd8a3e9cec4be87ed92062252fff155f3e2

But it might mean that in the original tests, the failed CIs may use `wasm_runtime_invoke_native` instead of quick-aot-entry to run spec cases, I am not sure whether there are errors in handling riscv ABI in `wasm_runtime_invoke_native`:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/7553863769/job/20568614964
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/7553863769/job/20565649590

**Pull request merged**